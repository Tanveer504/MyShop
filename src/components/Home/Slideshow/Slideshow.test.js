import React from 'react';
import Slideshow from './Slideshow';
import ReactDom from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import {render, screen} from '@testing-library/react';
import renderer from 'react-test-renderer'

describe("Slideshow Component", () => {
    it('renders the component properly', () => {
        const divEl = document.createElement('div');
        ReactDom.render(<BrowserRouter><Slideshow/></BrowserRouter>, divEl);
    });

    it('should have products as text', () => {
        render(<BrowserRouter><Slideshow/></BrowserRouter>);
        const productsText = screen.getByText('BROWSE 1000+ PRODUCTS');
        expect(productsText).toBeInTheDocument();
    });

    it('should have right Slideshow comp snapshot', () => {
        const tree = renderer.create(<BrowserRouter><Slideshow/></BrowserRouter>).toJSON();
        expect(tree).toMatchSnapshot();
    });
  
   
})