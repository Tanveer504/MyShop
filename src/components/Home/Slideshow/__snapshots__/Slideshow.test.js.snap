// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slideshow Component should have right Slideshow comp snapshot 1`] = `
<section>
  <div
    className="slideshowSection"
  >
    <div
      className="slideShowBox homePageImage"
    >
      <img
        alt="deliveryimage"
        className="homeImage"
        height="80%"
        src="del.jpg"
        width="80%"
      />
    </div>
    <div
      className="slideShowBox"
    >
      <h1
        className="slideshowHeading"
      >
        Same Day Delivery
      </h1>
      <p
        className="slideshowContent"
      >
        Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using Content here, content here, making it look like readable English. Many desktop publishing packages and web page.
      </p>
      <div
        className="browserButton"
      >
        <a
          className="products"
          href="/products"
          onClick={[Function]}
        >
          <b>
            BROWSE 1000+ PRODUCTS
          </b>
        </a>
      </div>
      <button
        className="arrow arrowRight"
        onClick={[Function]}
      />
      <button
        className="arrow arrowLeft"
        onClick={[Function]}
      />
    </div>
  </div>
</section>
`;
